docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1271 @@
+- Calls `setDivider()` to format the output.
+
+#### `enterCommandMessage()`
+
+Prompts the user to enter a command.
+
+- Outputs: "Enter command: "
+
+#### `emptyInputMessage()`
+
+Handles cases where the user provides no input.
+
+- Outputs: "No input detected. Exiting program..."
+
+#### `setDivider()`
+
+Displays a horizontal divider for visual separation.
+
+- Outputs: A line of 80 underscores.
+
+#### `errorMessageTag()` _(static)_
+
+Returns the error message prefix.
+
+- Returns: "ERROR: "
+
+#### `invalidIndexMessage()`
+
+Displays an error message for an invalid expense index.
+
+- Returns: "Invalid expense index."
+
+#### `displayExitMessage()`
+
+Displays a farewell message when the user exits the application.
+
+- Outputs: "Thank you for using the Expense Manager. Goodbye!"
+
+#### `displayInvalidCommandMessage()`
+
+Displays an error message for invalid user commands.
+
+- Outputs: "Invalid command."
+
+#### `createNewFileMessage()` _(static)_
+
+Notifies the user when a new file is created.
+
+- Outputs: "Created a new file!"
+
+#### `displayEmptyGroupMessage()`
+
+Displays a message when attempting to access a group with no members.
+
+- Returns: "No members in this group."
+
+#### `displayMissingFileExceptionMessage()`
+
+Displays an error message when the owed amounts file is missing.
+
+- Outputs: "Owed amounts file not found. No amounts to display."
+
+#### `displayInvalidAmountExceptionMessage()`
+
+Handles errors when parsing owed amounts.
+
+- Outputs: "Error parsing owed amounts. Some amounts may not be displayed."
+
+#### `displayMissingGroupMessage()`
+
+Displays an error message when the specified group is not found.
+
+- Outputs: "Group not found."
+
+#### `displayCommandList()`
+
+Displays the list of available commands along with their descriptions and usage examples.
+
+- Prints a formatted command list to the console.
+- Groups related commands for clarity.
+- Provides details about user input expectations.
+
+### 3.2.3 Summary Class
+
+The Summary class manages the visualization and analysis of expense data. It provides comprehensive analytics through different views and formats.
+
+#### Key Features:
+
+- Generates monthly expense summaries with trend analysis
+- Creates category-wise breakdowns of expenses
+- Calculates percentage distributions across expense categories
+- Provides comparative analysis between different time periods
+- Supports data export functionality for external analysis
+- Integrates with ExpenseClassifier for accurate categorization
+- Maintains historical data for trend analysis
+
+#### Implementation:
+
+- Uses Java's built-in charting libraries for visualization
+- Implements data aggregation algorithms for summary generation
+- Provides both textual and graphical representation options
+- Maintains data integrity through proper validation
+- Supports multiple currency representations
+
+![SummarySequenceDiagram.png](diagrams/SummarySequenceDiagram.png)
```

Comment:
The diagram is a little hard to read. If possible and if time permits, try to make it more readable.